src/python/dxpy/scripts/dx.py

```diff
@@ -2513,22 +2513,16 @@ def install(args):
 
 def uninstall(args):
     app_desc = get_app_from_path(args.app)
-    if app_desc is None:
-        try:
-            dxpy.api.app_uninstall(args.app)
-            print('Uninstalled application with id: ' + args.app)
-        except:
-            try:
-                dxpy.api.app_uninstall('app-' + args.app)
-                print('Uninstalled application with name: ' + args.app)
-            except:
-                err_exit()
+    if app_desc:
+        try_call(dxpy.api.app_uninstall, app_desc['id'])
+        print('Uninstalled the {name} app'.format(name=app_desc['name']))
     else:
-        try:
-            dxpy.api.app_uninstall(app_desc['id'])
-            print('Uninstalled the ' + app_desc['name'] + ' app')
-        except:
-            err_exit()
+        user_data = dxpy.api.user_describe(dxpy.whoami(), {"fields": {"appsInstalled": True}})
+        if args.app in user_data['appsInstalled']:
+            try_call(dxpy.api.app_uninstall, 'app-' + args.app)
+        else:
```

Comment:
Seems api doesn't add 'app-' prefix automatically. In case executing:
```
if args.app in user_data['appsInstalled']:
            try_call(dxpy.api.app_uninstall, args.app)
```
we will get
```
> dx describe user-000000000000000000000001 | grep Apps
Apps installed      identityFn, uninstall_test
> dx uninstall uninstall_test
ResourceNotFound: The specified URL could not be found, code 404
```